core/src/main/java/uk/ac/bbsrc/tgac/miso/core/security/util/LimsSecurityUtils.java

```diff
@@ -72,24 +81,38 @@ public static User fromLdapUser(InetOrgPerson details) {
     // user and this user will inherit the already-persisted userID.
     final UserImpl user = new UserImpl();
 
-    user.setActive(details.isAccountNonExpired());
-    user.setAdmin(details.getAuthorities().contains(new GrantedAuthorityImpl("ROLE_ADMIN")));
-    user.setExternal(details.getAuthorities().contains(new GrantedAuthorityImpl("ROLE_EXTERNAL")));
-    user.setInternal(details.getAuthorities().contains(new GrantedAuthorityImpl("ROLE_INTERNAL")));
-    user.setLoginName(details.getUsername().toLowerCase());
-    user.setPassword(details.getPassword());
-    user.setFullName(details.getDisplayName());
-    user.setEmail(details.getMail());
-
     final List<String> roles = new ArrayList<String>();
     for (final GrantedAuthority ga : details.getAuthorities()) {
-      roles.add(ga.toString());
+      roles.add(removePrefix(ga.toString(), LimsSecurityUtils.rolePrefix));
     }
     user.setRoles(roles.toArray(new String[0]));
 
+    user.setActive(details.isAccountNonExpired());
+    user.setAdmin(roles.contains(new GrantedAuthorityImpl("ROLE_ADMIN")));
```

Comment:
I've made this change. Unfortunately I forgot about making it a separate commit so that the changes would be obvious.